test/pylib/util.py

```diff
@@ -76,5 +78,21 @@ async def get_hosts() -> Optional[list[Host]]:
 
     return hosts
 
+def read_last_line(file_path: pathlib.Path, max_line_bytes: Optional[int] = 512):
+    file_size = os.stat(file_path).st_size
+    with file_path.open('rb') as f:
+        f.seek(max(0, file_size - max_line_bytes), os.SEEK_SET)
```

Comment:
If `max_line_bytes == None`, this will throw
```
TypeError: unsupported operand type(s) for -: 'int' and 'NoneType'
```

I don't think you want `max_line_bytes: Optional[int]`, you want `max_line_bytes: int` - you never pass `None` to this function.

Reply:
I can fix this during merge.

Reply:
fixed

Reply:
I should have moved it first, now this commit contains this change instead of just moving...